Tests/FluentKitTests/QueryBuilderTests.swift

```diff
@@ -0,0 +1,68 @@
+@testable import FluentKit
+@testable import FluentBenchmark
+import XCTest
+import Foundation
+import FluentSQL
+
+final class QueryBuilderTests: XCTestCase {
+    func testFirstEmptyResult() throws {
+        let db = DummyDatabaseForTestQueries(mockResults: [[]])
+
+        let retrievedPlanet = try Planet.query(on: db).first().wait()
+
+        XCTAssertNil(retrievedPlanet)
+    }
+
+    func testFirstSingleResult() throws {
+        let planet = Planet(id: 10, name: "Tully", galaxyID: 1)
+        let db = DummyDatabaseForTestQueries(
+            mockResults: [
+                [
+                    DummyDatabaseForTestQueries.DummyRow(
+                        dummyDecodedFields: [
+                            "id": planet.id as Any,
+                            "name": planet.name,
+                            "galaxy_id": planet.$galaxy.id
+                        ]
+                    )
+                ]
+            ]
+        )
+
+        let retrievedPlanet = try Planet.query(on: db).first().wait()
+
+        XCTAssertEqual(planet.id, retrievedPlanet?.id)
+        XCTAssertEqual(planet.name, retrievedPlanet?.name)
+        XCTAssertEqual(planet.$galaxy.id, retrievedPlanet?.$galaxy.id)
+    }
+
+    func testFirstManyResults() throws {
+        let planet = Planet(id: 10, name: "Tully", galaxyID: 1)
+        let db = DummyDatabaseForTestQueries(
+            mockResults: [
```

Comment:
This large initializer is kind of hard to read. I wonder if something like this would work:

```swift
let db = TestDatabase()
db.append(queryResult: [
   TestRow(["id": ...]),
   TestRow(["id": ...]), 
])
```

Reply:
It surely would work. Happy to move that direction. These days I always land on a more declarative look to begin with, but I can't say I find one more appealing than the other by a large margin.